### PR TITLE
Add Resource API to ballerina

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will add Resource APIs to Ballerina.

Resource API design in [1].

Resolves: https://github.com/ballerina-platform/ballerina-lang/issues/18881 https://github.com/ballerina-platform/ballerina-lang/issues/22767 https://github.com/ballerina-platform/ballerina-lang/issues/22296

[1] https://docs.google.com/document/d/1jB9-qnE31zIsSI4wuu95eisrP0dXulLI_YV1h22LhMc/edit?usp=sharing

### CODE REVIEW

File: stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java [hunk 1]
```diff
@@ -136,6 +147,56 @@ public static Object openReadableFile(String pathUrl) {
...
+            ReadableByteChannel readableByteChannel = Channels.newChannel(resourceAsStream);
+            Channel channel = new BlobIOChannel(new BlobChannel(readableByteChannel));
+            return createChannel(channel);
+        } catch (IOException e) {
+            return IOUtils.createError("Error unzipping file: " + executablePath);
+        }
+    }
+
+    public static ArrayValue getResources() {
+        ArrayList<String> list = new ArrayList<>();
```
Comment (keizer619) [inline stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java:175]: Can't we declare this as a List<String> type?

File: stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java [hunk 2]
```diff
@@ -136,6 +147,56 @@ public static Object openReadableFile(String pathUrl) {
         return channel;
     }
 
+    public static Object getResource(String pathUrl) {
```
Comment (keizer619) [inline stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java:150]: Shall we include java doc comments for public methods?

File: tests/jballerina-integration-test/src/test/resources/resources/Ballerina.toml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[project]
+org-name= "ballerina-test"
+version= "0.1.0"
+
+[dependencies]
```
Comment (keizer619) [inline tests/jballerina-integration-test/src/test/resources/resources/Ballerina.toml:5]: just wondering whether we can remove this if we don't have dependencies.

File: stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java (hunk 1 above)
Comment (praveennadarajah) [reply stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java]: Yes, we can declare this as List type and this is fixed.

File: stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java (hunk 2 above)
Comment (praveennadarajah) [reply stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/ByteChannelUtils.java:157]: Java doc comments added to public methods.

**Pull request closed without merging**